### PR TITLE
NEOS-934: added in support for generate js transformers

### PR DESCRIPTION
**Pull request by evisdrenova**
Implements generate_javascript transformers for both data gen and sync jobs. @nickzelei benthos doesn't like it when you use a `generate` input and don't provide a mapping, so I added a check on the mutations to see if they're empty and if so, default the mutation to an empty object in the `benthos-builder.go`.

```
	// for the generate input, benthos requires a mapping, so falling back to a
		// generic empty object if the mutations are empty
		if mutations == "" {
			mutations = "root = {}"
		}
```

The use-case is if you have all custom generate_javascript transformers and no regular mappings. LMK if you have a better idea on this.

### CODE REVIEW

File: backend/services/mgmt/v1alpha1/transformers-service/system_transformers.go [hunk 1]
```diff
@@ -505,6 +505,17 @@ var (
 				},
 			},
 		},
+		{
+			Name:        "Generate Javascript",
+			Description: "Write custom Javascript to generate synthetic data. ",
```
Comment (nickzelei) [inline backend/services/mgmt/v1alpha1/transformers-service/system_transformers.go:510]:
```suggestion
			Description: "Write custom Javascript to generate synthetic data.",
```

File: frontend/apps/web/app/(mgmt)/[account]/transformers/Sheetforms/GenerateeJavascriptForm.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+import { useTheme } from 'next-themes';
+import { ReactElement, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import {
+  IsUserJavascriptCodeValid,
+  ValidCode,
+} from '../../new/transformer/UserDefinedTransformerForms/UserDefinedTransformJavascriptForm';
+import { TRANSFORMER_SCHEMA_CONFIGS } from '../../new/transformer/schema';
+import { TransformerFormProps } from './util';
+interface Props extends TransformerFormProps<TransformJavascript> {}
```
Comment (nickzelei) [inline frontend/apps/web/app/(mgmt)/[account]/transformers/Sheetforms/GenerateeJavascriptForm.tsx:28]:
```suggestion
interface Props extends TransformerFormProps<GenerateJavascript> {}
```

File: frontend/apps/web/app/(mgmt)/[account]/transformers/Sheetforms/GenerateeJavascriptForm.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+export default function GenerateJavascriptForm(props: Props): ReactElement {
+  const { existingConfig, onSubmit, isReadonly } = props;
+
+  const form = useForm({
+    mode: 'onChange',
+    resolver: yupResolver(TRANSFORMER_SCHEMA_CONFIGS.transformJavascriptConfig),
```
Comment (nickzelei) [inline frontend/apps/web/app/(mgmt)/[account]/transformers/Sheetforms/GenerateeJavascriptForm.tsx:35]:
```suggestion
    resolver: yupResolver(TRANSFORMER_SCHEMA_CONFIGS.generateJavascriptConfig),
```

File: worker/pkg/workflows/datasync/activities/gen-benthos-configs/benthos-builder.go [hunk 4]
```diff
@@ -1607,14 +1630,45 @@ func shouldProcessFkColumn(t *mgmtv1alpha1.JobMappingTransformer) bool {
...
+// 	if jsCode != "" {
+// 		return fmt.Sprintf(`
+// function fn_%s(){
+//   %s
+// };
+// `, col, jsCode)
+// 	} else {
+// 		return ""
+// 	}
+// }
```
Comment (nickzelei) [inline worker/pkg/workflows/datasync/activities/gen-benthos-configs/benthos-builder.go:1655]: if this isnt needed let's get this comment block out of here.

File: worker/pkg/workflows/datasync/activities/gen-benthos-configs/benthos-builder.go [hunk 5]
```diff
@@ -1635,8 +1689,15 @@ benthos.v0_msg_set_structured(output);
 	return jsCode
 }
 
-func constructBenthosOutput(col string) string {
-	return fmt.Sprintf(`output["%[1]s"] = fn_%[1]s(input["%[1]s"], input);`, col)
+func constructBenthosOutput(col string, source mgmtv1alpha1.TransformerSource) string {
```
Comment (nickzelei) [inline worker/pkg/workflows/datasync/activities/gen-benthos-configs/benthos-builder.go:1692]: Can we update this function name to be more specific to javascript? The name is very generic, but it looks like it's used to build the javascript output object, not the benthos output.

File: worker/pkg/workflows/datasync/activities/gen-benthos-configs/benthos-builder.go [hunk 6]
```diff
@@ -737,25 +737,41 @@ func buildBenthosGenerateSourceConfigResponses(
...
 
+		mutations, err := buildMutationConfigs(ctx, transformerclient, tableMapping.Mappings, columnInfo)
+		if err != nil {
+			return nil, err
+		}
+
+		// for the generate input, benthos requires a mapping, so falling back to a
+		// generic empty object if the mutations are empty
+		if mutations == "" {
+			mutations = "root = {}"
```
Comment (nickzelei) [inline worker/pkg/workflows/datasync/activities/gen-benthos-configs/benthos-builder.go:753]: I think this makes sense here to accomodate the case where no generators have been supplied save for javascript ones.

**Review by nickzelei — COMMENTED** [review]
left a few minor comments.

File: worker/pkg/workflows/datasync/activities/gen-benthos-configs/benthos-builder.go (hunk 4 above)
Comment (evisdrenova) [reply worker/pkg/workflows/datasync/activities/gen-benthos-configs/benthos-builder.go:1655]: oops good catch, weird that the linter didn't catch that, but removed

**Pull request merged**